### ISSUE TITLE
Accessing ProximityPlacementGroup throws NullReferenceException

### ISSUE THREAD

**Issue by jonstelly** [Azure/azure-libraries-for-net#892]
**Describe the bug**
ProximityPlacementGroup() calls throw NullReferenceException for Resources without PPGs

***Exception or Stack Trace***

```
System.NullReferenceException : Object reference not set to an instance of an object.
   at Microsoft.Azure.Management.Compute.Fluent.AvailabilitySetImpl.ProximityPlacementGroup()
   at Microsoft.Azure.Management.Compute.Fluent.AvailabilitySetImpl.Microsoft.Azure.Management.Compute.Fluent.IAvailabilitySet.get_ProximityPlacementGroup()
```

**To Reproduce**
Access the ProximityPlacementGroup property on IAvailabiltySet, IVirtualMachine, IVirtualMachineScaleSet

***Code Snippet***
```
IAzure az = ...;
var ppg = az.AvailabilitySets.List().First().ProximityPlacementGroup;
```

**Expected behavior**
For the ProximityPlacementGroup property to return null instead of throwing an exception.

**Setup (please complete the following information):**
 - OS: Windows 10
 - IDE : VS 2019
 - NuGet version: 1.27.2

**Additional context**
The existing boilerplate code seems to look like this in AvailabilitySetImpl, VirtualMachineImpl, VirtualMachineScaleSetImpl:

```
public IProximityPlacementGroup ProximityPlacementGroup()
{
    ResourceId id = ResourceId.FromString(Inner.ProximityPlacementGroup.Id);

    ProximityPlacementGroupInner plgInner = Microsoft.Azure.Management.ResourceManager.Fluent.Core.Extensions.Synchronize(() => this.Manager.Inner.ProximityPlacementGroups.GetAsync(this.ResourceGroupName, id.Name));
    if (plgInner == null)
    {
        return null;
    }
    else
    {
        return new ProximityPlacementGroupImpl(plgInner);
    }
}
```

I think it just needs a check on the `Inner.ProximityPlacementGroup`, something like this:

```
public IProximityPlacementGroup ProximityPlacementGroup()
{
    if(Inner.ProximityPlacementGroup == null)
        return null;
    ResourceId id = ResourceId.FromString(Inner.ProximityPlacementGroup.Id);

    ProximityPlacementGroupInner plgInner = Microsoft.Azure.Management.ResourceManager.Fluent.Core.Extensions.Synchronize(() => this.Manager.Inner.ProximityPlacementGroups.GetAsync(this.ResourceGroupName, id.Name));
    if (plgInner == null)
    {
        return null;
    }
    else
    {
        return new ProximityPlacementGroupImpl(plgInner);
    }
}
```
**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by yaohaizh** [comment]
@ChenTanyi , could you help take a look at this issue to see whether we can fix this quickly? Thanks.

**Comment by ChenTanyi** [comment]
Thanks @jonstelly , will fix in 1.28.1
